### PR TITLE
[12.0] Add module project_iteration

### PR DESCRIPTION
**Pull request by ddufresne**
This module was created for version 11.0 in 2018.
It was moved from a private Numigi repository to github.com/numigi/odoo-project-addons
when migrating to version 12.0.

https://isidor.numigi.net/web#id=8257&view_type=form&model=project.task&action=292&active_id=337&menu_id=200

### CODE REVIEW

**Comment by ddufresne** [comment]
@foutoucour la seule chose que j'ai modifié est l'ajustement des traductions.

Sinon, j'ai amélioré le README.

J'ai aussi fusionné le module suivant avec celui-ci pour simplifier. Il s'agit d'un seul fichier javascript.

https://github.com/Numigi/odoo-web-addons/tree/12.0/.unported_addons/web_search_input_many2many

Ce module n'avait pas d'autre utilité et je doute qu'il puisse être utilisé éventuellement pour autre chose.

File: project_iteration/static/src/search_input_many2many.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    Without this module, the ID of the project is displayed in the search bar facet
+    instead of its display name.
+*/
+odoo.define("web_search_input_many2many.search_view_field", function(require) {
+"use strict";
+
+var searchInputs = require("web.search_inputs");
+
+searchInputs.Field.include({
+    facet_for(value){
```
Comment (ddufresne) [inline project_iteration/static/src/search_input_many2many.js]: Codacy Issue found: [Identifier 'facet_for' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3300066)

Comment (ddufresne) [inline project_iteration/static/src/search_input_many2many.js]: Codacy Issue found: [Method 'facet_for' has a complexity of 5. (complexity)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3300066)

File: project_iteration/static/src/search_input_many2many.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    Instead of filtering by a many2one field (project_id) like in vanilla Odoo,
+    the view is filtered by a many2many field (parent_project_ids).
+
+    This allows to display the tasks below a parent projects and its iterations.
+
+    Without this module, the ID of the project is displayed in the search bar facet
+    instead of its display name.
+*/
+odoo.define("web_search_input_many2many.search_view_field", function(require) {
```
Comment (ddufresne) [inline project_iteration/static/src/search_input_many2many.js]: Codacy Issue found: ['odoo' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3300066)

File: project_iteration/static/src/search_input_many2many.js (hunk 1 above)
Comment (ddufresne) [inline project_iteration/static/src/search_input_many2many.js]: Codacy Issue found: [Identifier 'facet_for' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3300066)

Comment (ddufresne) [inline project_iteration/static/src/search_input_many2many.js]: Codacy Issue found: [Method 'facet_for' has a complexity of 5. (complexity)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3300066)

File: project_iteration/static/src/search_input_many2many.js (hunk 2 above)
Comment (ddufresne) [inline project_iteration/static/src/search_input_many2many.js]: Codacy Issue found: ['odoo' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3300066)

**Review by foutoucour — APPROVED** [review]

**Pull request merged**